### DISCUSSION TITLE
How to return charts?

### DISCUSSION THREAD

**Discussion by istrupin** [sinaptik-ai/pandas-ai#1479] [Q&A] [answered]
### System Info

OS Version: MacOS Sequoia
Python Version: 3.11
pandasai version: 2.3.0

### 🐛 Describe the bug

When making a query to plot a chart with the following config:
```
{
        "verbose": True,
        "llm": _get_llm(),
        "open_charts": False,
        "enable_cache": False,
        "save_charts": True,
        "save_logs": False,
}
```
The response I see is:
```
Unfortunately, I was not able to answer your question, because of the following error:\n\nNo result returned\n
```
Despite the fact that a chart is indeed saved in the export folder.

The [relevant documentation](https://docs.pandas-ai.com/examples#plotting) shows that the output should state the name of the created chart file, which would be super helpful.

**Comment by gdcsinaptik** [comment]
@istrupin could you please share more info on what's the code you are using and the user query? 
Thanks a lot

**Comment by istrupin** [comment, marked as answer]
Hi @gdcsinaptik thank you so much for your response here.

So in playing around with this, I've found that the operative word is "plot".  I've been using "graph" and "chart" in my prompts (and as the output type given to the agent.chat method), and they were not consistently working, however "plot" seems to be doing the trick.

For the record, I'm using GPT 4o in azure.

The relevant code is pretty straightforward:

```
def query_lake(query, filenames, response_parser):
    lake = _create_lake_agent(filenames, user_id, tenant_id)
    agent_response = lake.chat(query=query, output_type=response_parser)
    agent_explanation = lake.explain()

def _create_lake_agent(files, user_id, tenant_id, response_parser_type=OutputType.STRING):
    dfs = _get_dataframes_from_files(files)
    config = {
        "verbose": True,
        "llm": _get_llm(),
        "open_charts": False,
        "enable_cache": False,
        "save_charts": True,
        "save_logs": False,
    }
    if response_parser_type == OutputType.DATAFRAME:
        config["response_parser"] = PandasDataFrame
    return Agent(dfs=dfs, config=config)  # type: ignore
```

For my current testing, my query is simply "Plot the prices over time".  This seems to do be working!

**Reply by gdcsinaptik** [reply]
Thanks for sharing this insight with the community :)
